forms-flow-web/src/components/CustomComponents/settingsModal.js

```diff
@@ -0,0 +1,351 @@
+    const inputValue = e.target.value;
+    setRoleInputEdit(inputValue);
+    setFilteredRolesEdit(userRoles.filter((role) =>
+      role.name.toLowerCase().includes(inputValue.toLowerCase())));
+  };
+
+  const handleRoleInputChangeCreate = (e) => {
+    const inputValue = e.target.value;
+    setRoleInputCreate(inputValue);
+    setFilteredRolesCreate(userRoles.filter((role) =>
+      role.name.toLowerCase().includes(inputValue.toLowerCase())));
+  };
+
+  const handleRoleInputChangeView = (e) => {
+    const inputValue = e.target.value;
+    setRoleInputView(inputValue);
+    setFilteredRolesView(userRoles.filter((role) =>
+      role.name.toLowerCase().includes(inputValue.toLowerCase())));
+  };
+
+  const handleRoleSelect = (role, setSelectedRoles) => {
+    setSelectedRoles((prevRoles) =>
+      !prevRoles.includes(role.name) ? [...prevRoles, role.name] : prevRoles);
+    clearRoleInputs();
+  };
+
+  const clearRoleInputs = () => {
+    setRoleInputEdit('');
+    setRoleInputCreate('');
+    setRoleInputView('');
+    setFilteredRolesEdit([]);
+    setFilteredRolesCreate([]);
+    setFilteredRolesView([]);
+  };
+
+  const removeRole = (role, setSelectedRoles) => {
+    setSelectedRoles((prevRoles) => prevRoles.filter((r) => r !== role));
+  };
+
+  const copyPublicUrl = () => {
+    copyText(url)
+      .then(() => {
+        setCopied(true);
+        setTimeout(() => {
+          setCopied(false);
+        }, 3000);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  };
+
+  const handleFormPathChange = (e) => {
+    setNewPath(e.target.value);
+  };
+
+  const dropEditRef = useRef(null);
+  const dropCreateRef = useRef(null);
+  const dropViewRef = useRef(null);
+
+
+  const handleClickOutside = (event) => {
+    if (dropEditRef.current && !dropEditRef.current.contains(event.target)) {
+      setFilteredRolesEdit([]);
+    }
+    if (dropCreateRef.current && !dropCreateRef.current.contains(event.target)) {
+      setFilteredRolesCreate([]);
+    }
+    if (dropViewRef.current && !dropViewRef.current.contains(event.target)) {
+      setFilteredRolesView([]);
+    }
+  };
+
+  useEffect(() => {
+    document.addEventListener('mousedown', handleClickOutside);
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
+  }, []);
+
+  
+
+  return (
+    <Modal className="settings-modal" show={show} onHide={handleClose} dialogClassName="modal-50w" backdrop="static" >
+      <Modal.Header>
+        <Modal.Title>{t("Settings")}</Modal.Title>
+      </Modal.Header>
+      <Modal.Body className="p-0">
+        {/* Section 1: Basic */}
+        <div className='section'>
+          <h5 className='fw-bold'>{t("Basic")}</h5>
+          <FormInput value={newFormName} label={t("Name")} onChange={handleFormNameChange} dataTestid="form-name" ariaLabel={t("Form Name")} />
+
+          <Form.Group className='settings-input' controlId="descriptionInput">
```

Comment:
Please replace this textarea with the reusable component created by Shuhaib @Bonymol-aot

Reply:
Updated